tests/src/FunctionalJavascript/CorporateContactFormTest.php

```diff
@@ -0,0 +1,383 @@
+  }
+
+  /**
+   * Trigger topic add ajax and assert new fields.
+   */
+  protected function assertTopicAjax($delta = 0): void {
+    /** @var \Behat\Mink\WebAssert $assert */
+    $assert = $this->assertSession();
+    /** @var \Behat\Mink\Element\DocumentElement $page */
+    $page = $this->getSession()->getPage();
+
+    $topic_name_key = "corporate_fields[topics_fieldset][group][$delta][topic_name]";
+    $topic_email_key = "corporate_fields[topics_fieldset][group][$delta][topic_email_address]";
+
+    $page->fillField($topic_name_key, 'topic-' . $delta);
+    $page->fillField($topic_email_key, $delta . '-topic@email.com');
+
+    // Test add topic group.
+    $add_topic = $page->find('css', 'input[value="Add topic"]');
+    $this->assertNotEmpty($add_topic);
+    $add_topic->click();
+    $assert->assertWaitOnAjaxRequest();
+
+    $delta++;
+    $topic_name_key = "corporate_fields[topics_fieldset][group][$delta][topic_name]";
+    $topic_email_key = "corporate_fields[topics_fieldset][group][$delta][topic_email_address]";
+
+    $element = $page->findField($topic_name_key);
+    $this->assertNotEmpty($element);
+    $element = $page->findField($topic_email_key);
+    $this->assertNotEmpty($element);
+
+    // Assert fields are required.
+    $page->fillField($topic_name_key, '');
+    $page->fillField($topic_email_key, '');
+    $add_topic->click();
+    $assert->assertWaitOnAjaxRequest();
+    $assert->fieldExists($topic_name_key)->hasClass('error');
+    $assert->fieldExists($topic_email_key)->hasClass('error');
+    $assert->pageTextContains('Topic name field is required.');
+    $assert->pageTextContains('Topic email address(es) field is required.');
+
+    // Assert email validaiton.
+    $page->fillField($topic_email_key, 'not an email');
+    $add_topic->click();
+    $assert->assertWaitOnAjaxRequest();
+    $assert->fieldExists($topic_email_key)->hasClass('error');
+    $assert->pageTextContains('is an invalid email address.');
+
+    // Add valid second row of values.
+    $page->fillField($topic_name_key, 'topic-' . $delta);
+    $page->fillField($topic_email_key, $delta . '-topic@email.com');
+    $add_topic->click();
+    $assert->assertWaitOnAjaxRequest();
+
+    $delta++;
+    $topic_name_key = "corporate_fields[topics_fieldset][group][$delta][topic_name]";
+    $topic_email_key = "corporate_fields[topics_fieldset][group][$delta][topic_email_address]";
+
+    $element = $page->findField($topic_name_key);
+    $this->assertNotEmpty($element);
+    $element = $page->findField($topic_email_key);
+    $this->assertNotEmpty($element);
+
+    // Test remove topic group.
+    $remove_topic = $page->find('css', 'input[value="Remove topic"]');
+    $this->assertNotEmpty($remove_topic);
+    $remove_topic->click();
+    $assert->assertWaitOnAjaxRequest();
+
+    $element = $page->findField($topic_name_key);
+    $this->assertEmpty($element);
+    $element = $page->findField($topic_email_key);
+    $this->assertEmpty($element);
```

Comment:
Add assertion that the first two are still there.

Reply:
done